src/sentry/incidents/logic.py

```diff
@@ -77,6 +78,41 @@ def create_incident(
     return incident
 
 
+def update_incident_status(incident, status, user=None, comment=None):
+    """
+    Updates the status of an Incident and write an IncidentActivity row to log
+    the change. When the status is CLOSED we also set the date closed to the
+    current time and (todo) take a snapshot of the current incident state.
+    """
+    if incident.status == status.value:
+        # If the status isn't actually changing just no-op.
+        return
+    with transaction.atomic():
+        create_incident_activity(
+            incident,
+            IncidentActivityType.STATUS_CHANGE,
+            user=user,
+            value=status.value,
+            previous_value=incident.status,
+            comment=comment,
+        )
+        kwargs = {
+            'status': status.value,
+        }
+        if status == IncidentStatus.CLOSED:
+            kwargs['date_closed'] = timezone.now()
```

Comment:
Should this be able to be specified by the user?

Reply:
I don't think so, at least for the moment. I haven't seen any date pickers in the mocks for when a status changes.